staging/src/k8s.io/kube-scheduler/extender/v1/types_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// TestCompatibility verifies that the types in extender/v1 can be successfully encoded to json and decoded back, even when lowercased,
+// since these types were written around JSON tags and we need to enforce consistency on them now.
+// @TODO: v2 of these types should be defined with proper JSON tags to enforce this
```

Comment:
are you missing a sentence here? :sweat_smile:

Reply:
Aldo, prefer something like `TODO(#1234)` for tracking

Reply:
I don't think so, is there something you'd like me to add? Opened an issue for this here: https://github.com/kubernetes/kubernetes/issues/88634

Reply:
It's not clear what "this" means in `v2 of these types should be defined with proper JSON tags to enforce this`

Reply:
I would just reference the issue for making the API more consistent with other serialized APIs, and put details in that issue.

Reply:
Did a little of both, I agree we can discuss details in the issue